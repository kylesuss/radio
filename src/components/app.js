import React, { Component } from 'react'
import { compose } from 'redux'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import Logo from 'components/logo'
import Player from 'components/player'
import { findStationBySlug } from 'selectors/station'
import StyledApp from 'styled/app'
import StyledPage from 'styled/page'
import withKeyboardShortcuts from 'containers/keyboard-shortcuts'
import DocumentTitle from 'react-document-title'

const DEFAULT_TITLE = 'Fresh Transmission - Curated Internet Radio'

class App extends Component {
  get documentTitle () {
    const { activeStation, playerIsPlaying } = this.props

    return activeStation && playerIsPlaying
      ? `${activeStation.name} | ${DEFAULT_TITLE}`
      : DEFAULT_TITLE
  }

  render () {
    const { children } = this.props

    return (
      <StyledApp.Container>
        <DocumentTitle title={this.documentTitle} />

        <Logo />
        <Player />

        <StyledApp.Main>
          <StyledPage.Container>
            {children}
          </StyledPage.Container>
        </StyledApp.Main>
      </StyledApp.Container>
    )
  }
}

App.propTypes = {
  activeStation: PropTypes.shape({
    name: PropTypes.string
  }),
  playerIsPlaying: PropTypes.bool.isRequired
}

const mapStateToProps = (state) => ({
  activeStation: findStationBySlug(state.stations.items, state.player.activeStation),
  playerIsPlaying: state.player.isPlaying
})

export {
  App,
  DEFAULT_TITLE
}

export default compose(
  withKeyboardShortcuts,
  connect(mapStateToProps)
)(App)
