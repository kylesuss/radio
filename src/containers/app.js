import { connect } from 'react-redux'
import { findStationBySlug } from 'selectors/station'
import App from 'components/app'

const mapStateToProps = (state) => {
  return {
    playerIsPlaying: state.player.isPlaying,
    activeStation: findStationBySlug(state.stations.items, state.player.activeStation)
  }
}

export default connect(
  mapStateToProps,
  null
)(App)
