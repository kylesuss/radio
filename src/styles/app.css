@import 'variables.css';
@import 'animations.css';

.app {
  background: $white;
  padding: $spacing-common;
  margin-top: 6px;
  min-height: calc(100vh - 6px);
}

.app__header {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: 6px;
  background: linear-gradient(to right, $green, $blue, $green, $blue, $green, $blue);
  box-shadow: 0 1px 15px 0px rgba(0, 0, 0, .4);
  z-index: 1;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    bottom: 0;

  }

  &::before {
    animation: animateAppHeader 3000ms ease-out infinite;
    background: linear-gradient(to right, $blue, $green, $blue, $green, $blue, $green);
  }
}

.app__logo {
  position: fixed;
  width: 180px;
  height: 200px;
  padding: 4px;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 0;
    height: 0;
    border: 2px solid transparent;
    animation: animateAppLogoBackground 450ms $ease-out-back 250ms forwards;
  }

  &__inner {
    position: absolute;
    left: 4px;
    width: calc(100% - 8px);
    height: calc(100% - 8px);
    z-index: 1;
    animation: animateAppLogoInner 200ms ease-out 750ms forwards;
  }

  &__primary,
  &__secondary {
    opacity: 0;
    animation: animateAppLogo 500ms $ease-out-back 1000ms forwards;
  }

  &__primary {
    font-size: 3rem;
    background: linear-gradient(to right, $blue, $green, $blue);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
  }

  &__secondary {
    font-size: 1.275rem;
  }
}

.app__container {
  margin-left: calc(180px + $spacing-common);

  &--with-open-player {
    padding-bottom: 5rem;
  }
}
