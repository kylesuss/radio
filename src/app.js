/* global IS_SENTRY_ENABLED, SENTRY_CONFIG_URL */
import React from 'react'
import { render } from 'react-dom'
import Root from 'containers/root'
import Raven from 'raven-js'
import 'styles/global'

const renderApp = () => render(
  <Root />,
  document.getElementById('root')
)

if (IS_SENTRY_ENABLED === true) {
  Raven.config(SENTRY_CONFIG_URL).install()
  Raven.context(() => renderApp())
} else {
  renderApp()
}
