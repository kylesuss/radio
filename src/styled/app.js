import styled from 'styled-components'
import * as colors from 'styles/colors'
import * as positioning from 'styles/positioning'
import * as transitions from 'styles/transitions'
import * as shadow from 'styles/shadow'

const Container = styled.div`
  background: ${colors.LIGHT_GREY};
  margin-top: ${positioning.HEIGHT_HEADER_PX};
  min-height: calc(100vh - ${positioning.HEIGHT_HEADER_PX});
`

const Header = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: ${positioning.HEIGHT_HEADER_PX};
  background: linear-gradient(to right, ${colors.GREEN}, ${colors.BLUE}, ${colors.GREEN}, ${colors.BLUE}, ${colors.GREEN}, ${colors.BLUE});
  z-index: ${positioning.Z_INDEX_APP_HEADER};
`

const LeftColumn = styled.div`
  position: fixed;
  top: ${positioning.HEIGHT_HEADER_PX};
  left: 0;
  bottom: 0;
  width: ${positioning.WIDTH_LEFT_COLUMN_PX};
  box-shadow: ${shadow.SETTINGS_COMMON} ${props => props.isShowingStationPreview ? 'transparent' : colors.SHADOW};
  transition: box-shadow ${props => props.isShowingStationPreview ? '0ms' : transitions.LENGTH_COMMON_MS} ease-out ${props => props.isShowingStationPreview ? '0ms' : transitions.LENGTH_COMMON_MS};
  z-index: ${positioning.Z_INDEX_APP_LEFT_COLUMN};
  background: ${colors.PURE_BLACK};
`

const Main = styled.main`
  margin-left: ${positioning.WIDTH_LEFT_COLUMN_PX};
`

export default {
  Container,
  Header,
  LeftColumn,
  Main
}
