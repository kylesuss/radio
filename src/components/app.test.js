import React from 'react'
import { App, DEFAULT_TITLE } from './app'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

jest.mock('root', () => ({
  store: jest.fn()
}))

const props = {
  activeStation: null,
  playerIsPlaying: false,
}

test('the document title is the default title', () => {
  const wrapper = shallow(<App  {...props} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

test('it shows the default title when there is no active station playing', () => {
  const wrapper = shallow(<App {...props} />)
  expect(wrapper.instance().documentTitle).toEqual(DEFAULT_TITLE)
})

test('it shows the station title when there is an active station playing', () =>{
  const activeStation = { name: ':name' }
  const wrapper = shallow(<App {...props} playerIsPlaying activeStation={activeStation} />)
  expect(wrapper.instance().documentTitle).toEqual(`${activeStation.name} | ${DEFAULT_TITLE}`)
})
