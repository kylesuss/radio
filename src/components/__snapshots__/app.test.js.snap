// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the document title is the default title 1`] = `
<app__Container>
  <SideEffect(DocumentTitle)
    title="Fresh Transmission - Curated Internet Radio"
  />
  <Logo />
  <withRouter(Connect(Player)) />
  <app__Main>
    <page__Container />
  </app__Main>
</app__Container>
`;
